org.adempiere.base/src/org/compiere/model/PO_Record.java

```diff
@@ -184,80 +228,124 @@ private static KeyNamePair[] getTablesWithModelCascade(String tableName, String
 	 * If a referencing Record ID or Record UU exists to the deleted record, set it to NULL 
 	 * @param AD_Table_ID
-	 * @param Record_ID
+	 * @param Record_IDorUU record ID (int) or UUID (String)
 	 * @param trxName
 	 */
-	public static void setRecordIdNull(int AD_Table_ID, int Record_ID, String trxName){
-		KeyNamePair[] tables = getTablesWithRecordIDColumn(MColumn.FKCONSTRAINTTYPE_ModelSetNull, trxName);
+	public static void setRecordNull(int AD_Table_ID, Serializable Record_IDorUU, String trxName){
+		int refId;
+		String columnName;
+		if (Record_IDorUU instanceof Integer) {
+			refId = DisplayType.RecordID;
+			columnName = "Record_ID";
+		} else if (Record_IDorUU instanceof String) {
+			refId = DisplayType.RecordUU;
+			columnName = "Record_UU";
+		} else {
+			log.warning(Record_IDorUU.getClass().getName() + " not supported for ID/UUID");
+			return;
+		}
+		KeyNamePair[] tables = getTablesWithConstraintType(refId, MColumn.FKCONSTRAINTTYPE_ModelSetNull, trxName);
 		// Table loop
+		StringBuilder whereClause = new StringBuilder("AD_Table_ID=? AND ").append(columnName).append("=?");
 		for (KeyNamePair table : tables) {
-			List<PO> poList = new Query(Env.getCtx(), table.getName(), " AD_Table_ID = ? AND Record_ID = ? ", trxName)
-					.setParameters(AD_Table_ID, Record_ID)
+			List<PO> poList = new Query(Env.getCtx(), table.getName(), whereClause.toString(), trxName)
+					.setParameters(AD_Table_ID, Record_IDorUU)
 					.list();
 			
 			int count = 0;
 			for(PO po : poList) {
-				if (po.get_Table_ID() == AD_Table_ID && po.get_ID() == Record_ID)
+				if (   po.get_Table_ID() == AD_Table_ID
+					&& (   (Record_IDorUU instanceof Integer && po.get_ID() == (Integer) Record_IDorUU)
+						|| (Record_IDorUU instanceof String && Record_IDorUU.equals(po.get_UUID()))
+					   )
+				   )
 					continue;
-				if (po.isColumnMandatory(po.get_ColumnIndex("Record_ID")))
-					po.set_Value("Record_ID", 0);
-				else
-					po.set_Value("Record_ID", null);
+				if (po.isColumnMandatory(po.get_ColumnIndex(columnName))) {
+					if (Record_IDorUU instanceof Integer)
+						po.set_Value(columnName, 0);
+					else
+						po.set_Value(columnName, "");
+				} else {
+					po.set_Value(columnName, null);
+				}
 				po.saveEx(trxName);
 				count++;
 			}
 			if (count > 0) {
-				if (log.isLoggable(Level.CONFIG)) log.config(table.getName() + " (" + AD_Table_ID + "/" + Record_ID + ") #" + count);
+				if (log.isLoggable(Level.CONFIG)) log.config(table.getName() + " (" + AD_Table_ID + "/" + Record_IDorUU + ") #" + count);
 			}
 		}
 	}
 	
 	/**
 	 * 	An entry Exists for restrict table/record combination
 	 *	@param AD_Table_ID table
-	 *	@param Record_ID record
+	 *	@param Record_IDorUU record ID (int) or UUID (String)
 	 *	@param trxName transaction
 	 *	@return error message (Table Name) or null
 	 */
-	static String exists (int AD_Table_ID, int Record_ID, String trxName)
+	static String exists (int AD_Table_ID, Serializable Record_IDorUU, String trxName)

```

Comment:
this is using package visibility. better change it to either protected or public

Reply:
done